app/ts/simulation/simulator.ts

```diff
@@ -36,19 +43,53 @@ const logHandler = new Map<string, Loghandler>([
 	[ERC1155_TRANSFERSINGLE_LOG, handleERC1155TransferSingle],
 ])
 
-export const visualizeTransaction = (blockNumber: bigint, singleMulticallResponse: SingleMulticallResponse) => {
-	if (singleMulticallResponse.statusCode !== 'success') return { ethBalanceChanges: [], tokenResults: [], blockNumber }
-	let tokenResults: TokenVisualizerResult[] = []
-	for (const eventLog of singleMulticallResponse.events) {
-		const logSignature = eventLog.topics[0]
-		if (logSignature === undefined) continue
+const parseEvents = async (singleMulticallResponse: SingleMulticallResponse, ethereumClientService: EthereumClientService, userAddressBook: UserAddressBook): Promise<MaybeParsedEvents> => {
+	if (singleMulticallResponse.statusCode !== 'success' ) return []
+	return await Promise.all(singleMulticallResponse.events.map(async (event) => {
+		// todo, we should do this parsing earlier, to be able to add possible addresses to addressMetaData set
```

Comment:
Do we have a ticket for this so it doesn't get lost/forgotten?

Reply:
its done in later PR